azure-kusto-ingest/azure/kusto/ingest/exceptions.py

```diff
@@ -1,39 +1,39 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License
-
+from azure.kusto.data.data_format import DataFormat, IngestionMappingKind
 from azure.kusto.data.exceptions import KustoClientError
 
 
-class KustoMappingAndMappingReferenceError(KustoClientError):
+class KustoMappingError(KustoClientError):
     """
-    Error to be raised when ingestion properties include both
-    a mapping and a mapping reference
+    Raised when the provided mapping arguments are invalid.
     """
 
-    def __init__(self):
-        message = "Ingestion properties contain both an explicit mapping and a mapping reference."
-        super(KustoMappingAndMappingReferenceError, self).__init__(message)
-
 
 class KustoDuplicateMappingError(KustoClientError):
     """
-    Error to be raised when ingestion properties include two explicit mappings.
+    Raised when ingestion properties include both
+    column mappings and a mapping reference
     """
 
     def __init__(self):
-        message = "Ingestion properties contain two explicit mappings."
+        message = "Ingestion properties can't contain both an explicit mapping and a mapping reference."
         super(KustoDuplicateMappingError, self).__init__(message)
 
 
-class KustoMissingMappingReferenceError(KustoClientError):
+class KustoMissingMappingError(KustoClientError):
     """
-    Error to be raised when ingestion properties has data format of Json, SingleJson, MultiJson or Avro
-    but ingestion mappings reference was not defined.
+    Raised when ingestion properties has data format that requires a mapping and is not defined,
+    or when provided a mapping kind without a mapping.
     """
 
-    def __init__(self, data_format: str):
-        message = f"When stream format is {data_format}, mapping name must be provided."
-        super(KustoMissingMappingReferenceError, self).__init__(message)
+    @staticmethod
+    def required_data_format(data_format: DataFormat) -> "KustoMissingMappingError":
+        return KustoMissingMappingError(f"When stream format is '{data_format.kusto_value}', a mapping must be provided.")
+
+    @staticmethod
+    def required_with_kind(kind: IngestionMappingKind) -> "KustoMissingMappingError":
+        return KustoMissingMappingError(f"When ingestion mapping kind is set ('{kind.value}'), a mapping must be provided.")
```

Comment:
nit: I'm surprised to see hardcoded static methods within the exception class that return the exception constructed with predefined text.
I'd either expect the caller to provide the text, or for the caller to provide the text based on a constant (like https://github.com/Azure/azure-kusto-java/blob/master/data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java#L111).

I guess there are different approaches, so ignore if you prefer this approach. But if my experience is enough to go by (definitely might not be), this is a non-standard approach.